irida_staramr_results/api/irida_api.py

```diff
@@ -398,24 +400,30 @@ def get_analysis_result_files(self, analysis_submission_id):
             try:
                 file_url = self._get_file_url(analysis_submission_id, file_key)
             except exceptions.IridaKeyError:
-                """
-                Catches an exception if an analysis submission does not contain an analysis result.
-                For our case, this shouldn't happen since we use completed amr type analysis submission given 
-                by the caller (amr_downloader).
-                """
-                logging.error(f"No analysis result exists for analysis submission id "
-                             f"[{analysis_submission_id}]. Check analysis submission id [{analysis_submission_id}] "
-                             f"and ensure the analysis status is COMPLETED.")
+
+                if file_key == "staramr-pointfinder.tsv":
```

Comment:
I added some comment in the code to explain why we are skipping the error here.

Because PointFinder is optional in an amr analysis run, the `staramr-pointfinder.tsv` result file does not always exists. So, it is acceptable to ignore the error that is caused by the api call requesting a file url that does not exists.